Assets/SimpleAnimationComponent/SimpleAnimation.cs

```diff
@@ -71,6 +71,9 @@ public WrapMode wrapMode
 
     public void AddClip(AnimationClip clip, string newName)
     {
+        if (clip == null)
+            throw new System.NullReferenceException("clip");
```

Comment:
Indeed, that's the right place for this, but could you replace NullReferenceException (the object you are interacting with is null), by ArgumentNullException (the argument is null)?

Reply:
Ah, oops. Sure.